internal/graph/resolvers/composite.go

```diff
@@ -213,6 +218,7 @@ func (r *compositeResourceSpec) Resources(ctx context.Context, obj *model.Compos
 		out.Nodes = append(out.Nodes, kr)
 		out.TotalCount++
 	}
+	out.Nodes = out.Nodes[:out.TotalCount]
```

Comment:
I believe this can be dropped given you are just creating a slice with capacity of the length of references.